mwclient/client.py

```diff
@@ -979,7 +993,15 @@ def ask(self, query, title=None):
         while offset is not None:
             results = self.raw_api('ask', query='{query}|offset={offset}'.format(
                 query=query, offset=offset), http_method='GET', **kwargs)
-
-            offset = results.get('query-continue-offset')
-            for key, value in results['query']['results'].iteritems():
-                yield {key: value}
+            if results.get('error'):  # malformed query, maybe should raise an exception
```

Comment:
Agree that this should raise an exception. Could you try passing results through `handle_api_result()`, which will raise an exception if the results contains an error?